iidm/iidm-serde/src/main/java/com/powsybl/iidm/serde/extensions/BatteryShortCircuitSerDe.java

```diff
@@ -17,69 +16,52 @@
 import com.powsybl.iidm.network.extensions.BatteryShortCircuit;
 import com.powsybl.iidm.network.extensions.BatteryShortCircuitAdder;
 import com.powsybl.iidm.serde.IidmVersion;
-import com.powsybl.iidm.serde.NetworkDeserializerContext;
-import com.powsybl.iidm.serde.NetworkSerializerContext;
 
-import java.io.InputStream;
-import java.util.List;
-import java.util.Objects;
+import static com.powsybl.iidm.serde.extensions.BatteryShortCircuitSerDe.Version.V_1_0_LEGACY;
+import static com.powsybl.iidm.serde.extensions.BatteryShortCircuitSerDe.Version.V_1_0_LEGACY_2;
 
 /**
  * @author Coline Piloquet {@literal <coline.piloquet@rte-france.fr>}
  */
 @AutoService(ExtensionSerDe.class)
-public class BatteryShortCircuitSerDe extends AbstractVersionableNetworkExtensionSerDe<Battery, BatteryShortCircuit> {
+public class BatteryShortCircuitSerDe extends AbstractVersionableNetworkExtensionSerDe<Battery, BatteryShortCircuit, BatteryShortCircuitSerDe.Version> {
 
-    public static final String V1_0_LEGACY = "1.0-legacy";
-    public static final String V1_0_LEGACY_2 = "1.0-legacy-2";
-    private static final ImmutableSortedSet<String> LEGACY_VERSIONS = ImmutableSortedSet.<String>reverseOrder().add(V1_0_LEGACY, V1_0_LEGACY_2).build();
+    private static final ImmutableSortedSet<BatteryShortCircuitSerDe.Version> LEGACY_VERSIONS = ImmutableSortedSet.<BatteryShortCircuitSerDe.Version>reverseOrder().add(V_1_0_LEGACY, V_1_0_LEGACY_2).build();
 
-    public BatteryShortCircuitSerDe() {
-        super(BatteryShortCircuit.NAME, BatteryShortCircuit.class, "bsc",
-                new ImmutableMap.Builder<IidmVersion, ImmutableSortedSet<String>>()
-                        .put(IidmVersion.V_1_0, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_1, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_2, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_3, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_4, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_5, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_6, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_7, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_8, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_9, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_10, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_11, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_12, LEGACY_VERSIONS)
-                        .put(IidmVersion.V_1_13, ImmutableSortedSet.<String>reverseOrder().add(V1_0_LEGACY, V1_0_LEGACY_2, "1.0").build())
```

Comment:
Because of the `reverseOrder()` here and in the `LEGACY_VERSIONS`'s definition, `getVersion(IidmVersion)` should be redefined to keep the same behavior as previously:

```java
    @Override
    public Version getVersion(IidmVersion networkVersion) {
        if (IidmVersion.V_1_13.compareTo(networkVersion) > 0) {
            return V_1_0_LEGACY;
        } else if (networkVersion == IidmVersion.V_1_13) {
            return V_1_0;
        } else {
            return super.getVersion(networkVersion);
        }
    }
```